### PR TITLE
Adding Student To cohort Function

### PR DESCRIPTION
**Pull request by HemaSAli**
Relates #8

### CODE REVIEW

File: public/js/cohortStudent.js [hunk 1]
```diff
@@ -1,7 +1,65 @@
...
+const studentName = document.getElementById('studentName');
+const gitHubUserName = document.getElementById('gitHubUserName');
+const addStudentButton = document.getElementById('addStudentButton');
+const loading = document.getElementById('loading');
+const URL = window.location.href;
+const splitUrl = URL.split('/');
+const cohortID = splitUrl[splitUrl.length - 2];
+const deleteStudentButton = document.querySelectorAll('.delete');
 
 const addStudents = () => {
```
Comment (NouraldinS) [inline public/js/cohortStudent.js]: Give your function a name descriptive to it's task.

File: public/js/cohortStudent.js [hunk 2]
```diff
@@ -1,7 +1,65 @@
...
   addStd.classList.toggle('sectionAddstd--visible');
 };
 
+addStudentButton.addEventListener('click', (e) => {
+  e.preventDefault();
+
+  const studentNameValue = studentName.value;
+  const gitHubUserNameValue = gitHubUserName.value;
+  if (
+    studentNameValue.trim().length !== 0
```
Comment (NouraldinS) [inline public/js/cohortStudent.js]: `Boolean(studentNameValue.trim()) === Boolean(studentNameValue.trim().length === 0) // true`
Meaning that you don't have to specify the `.length !== 0` part, it is self explanatory to JavaScript.

File: public/js/cohortStudent.js [hunk 3]
```diff
@@ -1,7 +1,65 @@
...
+    const apiLink = `https://api.github.com/users/${gitHubUserNameValue}`;
+    fetch(apiLink, { method: 'GET' })
+      .then((loading.style.display = 'block'))
+      .then((response) => {
+        loading.style.display = 'none';
+        if (response.status !== 200) throw new Error('Invalid_username');
+        return response.json();
+      })
+      .then((response) => {
+        const { avatar_url, html_url } = response;
```
Comment (NouraldinS) [inline public/js/cohortStudent.js]:
```js
const { avatar_url: avatarUrl, html_url: htmlUrl } = response;
```
In order to stay consistent.

File: src/database/queries/addCohortStudent.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+const dbConnection = require('../db_connection');
+
+const addStudentQuery = (newStudent, cb) => {
+  const { name } = newStudent;
```
Comment (NouraldinS) [inline src/database/queries/addCohortStudent.js]:
```js
const { name, username, avatar_url: avatarUrl, html_url: htmlUrl, cohortId } = newStudent;
```

File: src/views/cohortStudents.hbs [hunk 5]
```diff
@@ -1,14 +1,14 @@
 <section class="content">
   <section class="sectionstd" id="student">
     <button class="button button2 buttonstd" onclick="addStudents()">ADD Student</button>
-    <hr> 
-    {{#if err}}
-      <h3>{{err}}</h3>
+    <hr> {{#if err}}
```
Comment (NouraldinS) [inline src/views/cohortStudents.hbs]: Fix indentation.

**Review by NouraldinS — CHANGES_REQUESTED** [review]

**Review by ali-7 — APPROVED** [review]

File: public/js/cohortStudent.js (hunk 1 above)
Comment (HemaSAli) [reply public/js/cohortStudent.js]: Done , thanks nour

File: public/js/cohortStudent.js (hunk 2 above)
Comment (HemaSAli) [reply public/js/cohortStudent.js]: new thing for me , thanks nour

File: public/js/cohortStudent.js (hunk 3 above)
Comment (HemaSAli) [reply public/js/cohortStudent.js]: thanks Nour  , it really made a problem for me

File: src/database/queries/addCohortStudent.js (hunk 4 above)
Comment (HemaSAli) [reply src/database/queries/addCohortStudent.js]: I edited it , by relating to previous solution

**Review by lubnaabd — APPROVED** [review]

**Pull request merged**